servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -439,15 +444,15 @@ boolean processPolicy(ObjectStoreConnection con, Policy originalPolicy, String d
 
             if (!ignoreDeletes) {
                 for (Assertion assertion : delAssertions) {
-                    if (!con.deleteAssertion(domainName, policyName, assertion.getId())) {
+                    if (!con.deleteAssertion(domainName, policyName, null, assertion.getId())) {
```

Comment:
The function processPolicy is only called from APIs that use the active version only.
However, to prevent bugs in case the code changes in the future and this function will be used for both active and disable policy versions, I'll change it as requested.

Reply:
@OferLevi85  That is not correct. processPolicy is called from executePutPolicy which is called from putPolicy. If I create a new version of the policy I can manage that version from either adding new assertions or just using putPolicy by providing the full object. In the putPolicy we don't have any restrictions where we check if you're modifying the active policy. So if the user uses that API to modify a non-active version, the code will update the active version which is not the user was doing. I'll merge the PR and will do a subsequent fix for this issue.